pkg/config/config.go

```diff
@@ -95,6 +95,29 @@ type Webhook struct {
 	BasicAuth HTTPBasicAuth       `json:"basicAuth,omitempty"`
 }
 
+type Addon struct {
+	Enabled       bool   `json:"enabled,omitempty"`
+	ValuesContent string `json:"valuesContent,omitempty"`
+}
+
+type LHDefaultSettings struct {
+	GuaranteedEngineManagerCPU  uint32 `json:"guaranteedEngineManagerCPU,omitempty"`
+	GuaranteedReplicaManagerCPU uint32 `json:"guaranteedReplicaManagerCPU,omitempty"`
```

Comment:
Sure, the new harvester-settings will be updated to https://docs.harvesterhci.io/dev/advanced/settings

Btw: with those new options, user can set them in both ISO and PXE installation, using additional harvester configuration YAML file.

This will be especially helpful in single-node cluster deployment:

(1) Users do not need to wait unti the cluser is ready and then set it via WEBUI, and as said, the cluster rebooting is required.

(2) Those prometheus/grafana used PVCs will be able to have only 1 replica during installation, thats not possbile before this PR. And the PVC is `healthy` instead of `degraded`.

(3) Strictly speaking, the 12% default value is arguable, I do not think we can convice all users with strong proof, we expose them and write them down in the doc, user has freedom to customize the values. Suppose users have already deployed lots of single-node clusters in their enviroment, they may have a best value for them, and they can set them during new cluster installation.

@guangbochen @bk201

Reply:
doc PR:
https://github.com/harvester/docs/pull/301